diff --git a/dist/runtime/entries/node-cluster.mjs b/dist/runtime/entries/node-cluster.mjs
index a5f976220de338103eb84604442312df904d7c96..b5b6eac0bb94dfb699dfaa0292acf60abaa94fd1 100644
--- a/dist/runtime/entries/node-cluster.mjs
+++ b/dist/runtime/entries/node-cluster.mjs
@@ -3,6 +3,7 @@ import cluster from "node:cluster";
 import { getGracefulShutdownConfig } from "../shutdown.mjs";
 import { trapUnhandledNodeErrors } from "../utils.mjs";
 function runMaster() {
+  let port = process.env.PORT
   const numberOfWorkers = Number.parseInt(process.env.NITRO_CLUSTER_WORKERS) || (os.cpus().length > 0 ? os.cpus().length : 1);
   for (let i = 0; i < numberOfWorkers; i++) {
     cluster.fork();
@@ -10,6 +11,7 @@ function runMaster() {
   let isShuttingDown = false;
   cluster.on("exit", () => {
     if (!isShuttingDown) {
+      process.env.PORT = port
       cluster.fork();
     }
   });
@@ -43,6 +45,11 @@ function runMaster() {
       process.once(signal, onShutdown);
     }
   }
+  process.env.PORT = 61111
+  import("./node-server.mjs").catch((error) => {
+    console.error(error);
+    process.exit(1);
+  });
 }
 function runWorker() {
   import("./node-server.mjs").catch((error) => {
